File: integraupt-web/Integraupt-Backend/BackendInicidencia/src/main/java/com/inicidencia/repository/IncidenciaRepository.java
package com.inicidencia.repository;

import com.inicidencia.dto.IncidenciaGestionRow;
import com.inicidencia.model.Incidencia;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface IncidenciaRepository extends JpaRepository<Incidencia, Integer> {
    List<Incidencia> findByReservaIdOrderByFechaReporteDesc(Integer reservaId);

    @Query("""
            SELECT new com.inicidencia.dto.IncidenciaGestionRow(
                i.id,
                r.id,
                u.id,
                u.nombre,
                u.apellido,
                u.numeroDocumento,
                e.id,
                e.codigo,
                e.nombre,
                esc.id,
                esc.nombre,
                fac.id,
                fac.nombre,
                i.descripcion,
                i.fechaReporte
            )
            FROM Incidencia i
            JOIN i.reserva r
            JOIN r.espacio e
            JOIN e.escuela esc
            JOIN esc.facultad fac
            LEFT JOIN r.usuario u
            WHERE (:escuelaId IS NULL OR esc.id = :escuelaId)
              AND (:facultadId IS NULL OR fac.id = :facultadId)
              AND (:espacioId IS NULL OR e.id = :espacioId)
              AND (
                    :search IS NULL OR :search = '' OR
                    LOWER(i.descripcion) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(e.nombre) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(e.codigo) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(COALESCE(u.nombre, '')) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(COALESCE(u.apellido, '')) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(CONCAT(COALESCE(u.nombre, ''), ' ', COALESCE(u.apellido, ''))) LIKE LOWER(CONCAT('%', :search, '%')) OR
                    LOWER(COALESCE(u.numeroDocumento, '')) LIKE LOWER(CONCAT('%', :search, '%'))
              )
            ORDER BY i.fechaReporte DESC
            """)
    List<IncidenciaGestionRow> buscarParaGestion(
            @Param("facultadId") Long facultadId,
            @Param("escuelaId") Long escuelaId,
            @Param("espacioId") Integer espacioId,
            @Param("search") String search
    );
}

File: Integraupt-Backend/servicio-incidencia/app/Repositories/IncidenciaRepository.php
<?php

namespace App\Repositories;

use Illuminate\Support\Facades\DB;
use App\Models\Incidencia;

class IncidenciaRepository
{
    public function findByReservaIdOrderByFechaReporteDesc($reservaId)
    {
        return Incidencia::where('Reserva', $reservaId)
            ->orderByDesc('FechaReporte')
            ->get();
    }

    public function buscarParaGestion($facultadId, $escuelaId, $espacioId, $search)
    {
        $query = DB::table('incidencia as i')
            ->join('reserva as r', 'r.IdReserva', '=', 'i.Reserva')
            ->join('espacio as e', 'e.IdEspacio', '=', 'r.espacio')
            ->join('escuela as esc', 'esc.IdEscuela', '=', 'e.Escuela')
            ->join('facultad as fac', 'fac.IdFacultad', '=', 'esc.IdFacultad')
            ->leftJoin('usuario as u', 'u.IdUsuario', '=', 'r.usuario')
            ->select(
                'i.IdIncidencia as id',
                'r.IdReserva as reservaId',
                'u.IdUsuario as usuarioId',
                'u.Nombre as usuarioNombre',
                'u.Apellido as usuarioApellido',
                'u.NumDoc as usuarioDocumento',
                'e.IdEspacio as espacioId',
                'e.Codigo as espacioCodigo',
                'e.Nombre as espacioNombre',
                'esc.IdEscuela as escuelaId',
                'esc.Nombre as escuelaNombre',
                'fac.IdFacultad as facultadId',
                'fac.Nombre as facultadNombre',
                'i.Descripcion as descripcion',
                'i.FechaReporte as fechaReporte'
            );

        if ($escuelaId !== null) {
            $query->where('esc.IdEscuela', $escuelaId);
        }

        if ($facultadId !== null) {
            $query->where('fac.IdFacultad', $facultadId);
        }

        if ($espacioId !== null) {
            $query->where('e.IdEspacio', $espacioId);
        }

        if ($search !== null && trim($search) !== '') {
            $searchTerm = '%' . strtolower(trim($search)) . '%';

            $query->where(function ($q) use ($searchTerm) {
                $q->whereRaw('LOWER(i.Descripcion) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(e.Nombre) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(e.Codigo) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(COALESCE(u.Nombre, "")) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(COALESCE(u.Apellido, "")) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(CONCAT(COALESCE(u.Nombre, ""), " ", COALESCE(u.Apellido, ""))) LIKE ?', [$searchTerm])
                    ->orWhereRaw('LOWER(COALESCE(u.NumDoc, "")) LIKE ?', [$searchTerm]);
            });
        }

        return $query
            ->orderByDesc('i.FechaReporte')
            ->get();
    }
}
